libraries/AP_Param/AP_Param.cpp

```diff
@@ -2005,6 +2005,16 @@ bool AP_Param::parse_param_line(char *line, char **vname, float &value, bool &re
     if (strlen(pname) > AP_MAX_NAME_SIZE) {
         return false;
     }
+
+#if CONFIG_HAL_BOARD == HAL_BOARD_SITL
+    // Workaround to prevent FORMAT_VERSION in param file resulting in invalid
+    // EEPROM. For details, see: https://github.com/ArduPilot/ardupilot/issues/15579
+    if (strncmp(pname, "FORMAT_VERSION", 14) == 0) {
```

Comment:
This is actually one place where "strcmp" might be better than strncmp - eliminates a constant and we're already doing string operations on pname further up.

Reply:
Ack. Updated.